deploy/local_single_user_deploy.sh

```diff
@@ -0,0 +1,218 @@
+#!/bin/bash
+
+# Variables ____________________________________________________________________________________________________________
+# Define some colors
+ERR='\033[1;31m\033[41m'
+INFO='\033[1,36m\033[42m'
+DEBUG='\033[0;33m'
+NC='\033[0m' # No Color
+NUM_KEEPERS=5
+
+# Directories
+WORK_DIR="/home/${USER}/chronolog"
+LIB_DIR="${WORK_DIR}/lib"
+CONF_DIR="${WORK_DIR}/conf"
+BIN_DIR="${WORK_DIR}/bin"
+
+# Files
+VISOR_BIN="${BIN_DIR}/chronovisor_server"
+KEEPER_BIN="${BIN_DIR}/chrono_keeper"
+CLIENT_BIN="${BIN_DIR}/client_lib_multi_storytellers"
+CONF_FILE="${CONF_DIR}/default_conf.json"
+
+# Methods ______________________________________________________________________________________________________________
+
+check_files() {
+    echo -e "${INFO}Checking required files...${NC}"
+    [[ ! -f ${VISOR_BIN} ]] && echo -e "${ERR}Visor binary file does not exist, exiting ...${NC}" && exit 1
+    [[ ! -f ${KEEPER_BIN} ]] && echo -e "${ERR}Keeper binary file does not exist, exiting ...${NC}" && exit 1
+    [[ ! -f ${CLIENT_BIN} ]] && echo -e "${ERR}Client binary file does not exist, exiting ...${NC}" && exit 1
+    [[ ! -f ${CONF_FILE} ]] && echo -e "${ERR}Configuration file does not exist, exiting ...${NC}" && exit 1
+    echo -e "${DEBUG}All required files are in place.${NC}"
+}
+
+generate_config_files() {
+    local num_files=$1
+    local bin_dir=$2
+    local default_conf=$3
+    local conf_dir=$4
+
+    # Check if default configuration file exists
+    if [ ! -f "$default_conf" ]; then
+        echo "Default configuration file $default_conf not found."
+        return 1
+    fi
+
+    # Extract initial ports from the configuration file
+    local base_port_keeper_record=$(jq '.chrono_keeper.KeeperRecordingService.rpc.service_base_port' "$default_conf")
+    local base_port_keeper_datastore=$(jq '.chrono_keeper.KeeperDataStoreAdminService.rpc.service_base_port' "$default_conf")
+
+    # Generate configuration files
+    for (( i=0; i<num_files; i++ ))
+    do
+        # Calculate new ports
+        local new_port_keeper_record=$((base_port_keeper_record + i))
+        local new_port_keeper_datastore=$((base_port_keeper_datastore + i))
+
+        # Create new config file name
+        local output_file="${conf_dir}/keeper_conf_$i.json"
+
+        # Use jq to modify the JSON
+        jq --arg bin_dir "$bin_dir" \
+           --argjson new_port_keeper_record $new_port_keeper_record \
+           --argjson new_port_keeper_datastore $new_port_keeper_datastore \
+           '.chrono_keeper.KeeperRecordingService.rpc.service_base_port = $new_port_keeper_record |
+            .chrono_keeper.KeeperDataStoreAdminService.rpc.service_base_port = $new_port_keeper_datastore |
+            .chrono_keeper.Logging.log.file = ($bin_dir + "/" + .chrono_keeper.Logging.log.file) |
```

Comment:
Personally prefer using `.log` extension as we have right now. It can trigger syntax highlighting automatically.

Reply:
They are currently generated with the extension defined in the configuration file. I think client default is still set to .txt but for me it generates as it is defined on the file. Do you want me to introduce something to force this to do that way?